index.js

```diff
@@ -82,35 +88,17 @@ function buildProcessTree (parentPid, tree, pidsToProcess, cb) {
             if (Object.keys(pidsToProcess).length == 0) {
                 cb();
             }
-            return
-        }
-
-        pids = [];
-        pid = '';
-        for (i = 0; i < allData.length; i++) {
-            if (allData[i] == '\n') {
-                pids.push(parseInt(pid, 10));
-                pid = '';
-                continue;
-            }
-            if (allData[i] != ' ') {
-                pid += allData[i];
-            }
+            return;
         }
 
-        pids.forEach(function (pid) {
-            tree[parentPid].push(pid)
-            tree[pid] = [];
-            pidsToProcess[pid] = 1;
-            buildProcessTree(pid, tree, pidsToProcess, cb);
+        allData.match(/\d+/g).forEach(function (pid) {
+          pid = parseInt(pid, 10);
+          tree[parentPid].push(pid);
+          tree[pid] = [];
+          pidsToProcess[pid] = 1;
+          buildProcessTree(pid, tree, pidsToProcess, spawnChildProcessesList, cb);
         });
     };
 
     ps.on('close', onClose);
 }
-
```

Comment:
So the need for these two functions is obsoleted by the ability to pass in `spawnChildProcessesList` correct?